src/System/Taffybar/ToggleMonitor.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE OverloadedStrings #-}
+module System.Taffybar.ToggleMonitor (
+  handleToggleRequests,
+  toggleableMonitors,
+  withToggleServer
+) where
+
+import           Control.Concurrent
+import qualified Control.Concurrent.MVar as MV
+import           Control.Monad
+import           Control.Monad.Trans
+import           Control.Monad.Trans.Maybe
+import qualified Data.Map as M
+import           Data.Maybe
+import qualified Graphics.UI.Gtk as Gtk
+import           Graphics.UI.Gtk.Gdk.Screen
+import           System.Taffybar
+import           Text.Read hiding (get, lift)
+import           Web.Scotty
+import           XMonad.Core ( whenJust )
+
+toggleableMonitors :: MV.MVar (M.Map Int Bool)
```

Comment:
The type `M.Map Int Bool` stands out here - is it encoding more information than an `IntSet`?  If so, I think I'd prefer a more expressive type to make it clear that there are three states (`True`, `False`, and "not present").